### PR TITLE
fix(lottery): `TooLowDelegationCountToDelegate` fails when funds withdrawing

### PR DESCRIPTION
**Pull request by Garandor**
Signed-off-by: Adam Reif <Garandor@manta.network>

## Description

---

Before we can approve this PR for merge, please make sure that **all** the following items have been checked off:
- [x] Connected to an issue with discussion and accepted design using zenhub "Connect issue" button below
- [x] Added **one** label out of the `L-` group to this PR
- [x] Added **one or more** labels from the `A-` and `C-` groups to this PR
- [x] Explicitly labelled `A-calamari` and/or `A-manta` if your changes are meant for/impact either of these (CI depends on it)
- [x] Re-reviewed `Files changed` in the Github PR explorer.

Situational Notes:
- If adding functionality, write unit tests!
- If importing a new pallet, choose a proper module index for it, and allow it in `BaseFilter`. Ensure **every** extrinsic works from front-end. If there's corresponding tool, ensure both work for each other.
- If needed, update our Javascript/Typescript APIs. These APIs are officially used by exchanges or community developers.
- If modifying existing runtime storage items, make sure to implement storage migrations for the runtime and test them with `try-runtime`. This includes migrations inherited from upstream changes, and you can search the diffs for modifications of `#[pallet::storage]` items to check for any.

### CODE REVIEW

**Review by ferrell-code — COMMENTED** [review]
same changes as #1218 in this pr but lgtm

**Comment by Garandor** [comment]
> same changes as #1218 in this pr but lgtm

yeah i built these sequentially, if 1218 is merged first the changeset just rebases

File: pallets/pallet-lottery/src/tests.rs [hunk 1]
```diff
@@ -332,10 +332,13 @@ fn unstaking_works_with_0_collators_left() {
                 Origin::signed(*ALICE),
                 balance * 2
             ));
-            assert_eq!(crate::StakedCollators::<Test>::iter().count(), 0);
+            assert_eq!(crate::StakedCollators::<Test>::iter().count(), 2);
+            assert_eq!(Lottery::withdrawal_request_queue().len(), 1);
             assert_ok!(Lottery::start_lottery(RawOrigin::Root.into()));
             roll_to_round_begin(3);
+            // assert_ok!(Lottery::draw_lottery(RawOrigin::Root.into()));
```
Comment (ferrell-code) [inline pallets/pallet-lottery/src/tests.rs]: commented out line

Comment (Garandor) [reply pallets/pallet-lottery/src/tests.rs]: thanks, fixed

**Review by ghzlatarev — APPROVED** [review]

**Review by ferrell-code — APPROVED** [review]

**Pull request merged**